src/api-client.ts

```diff
@@ -149,29 +155,33 @@ export class RevAiApiClient {
     }
 
     async getTranscriptObject(id: string): Promise<RevAiApiTranscript> {
-        try {
-            const response = await axios.get(`/jobs/${id}/transcript`, {
-                headers: { 'Accept': 'application/vnd.rev.transcript.v1.0+json' }
-            });
-            return response.data;
-        } catch (error) {
-            switch (error.response.status) {
-                case 401:
-                case 404:
-                    throw new RevAiApiError(error);
-                case 409:
-                    throw new InvalidStateError(error);
-                default:
-                    throw error;
-            }
-        }
+        return await this.getOutputsHelper(id, 'json', this.JSON_OUTPUT);
+    }
+
+    async getTranscriptObjectStream(id: string): Promise<Readable> {
+        return await this.getOutputsHelper(id, 'stream', this.JSON_OUTPUT);
     }
 
     async getTranscriptText(id: string): Promise<string> {
+        return await this.getOutputsHelper(id, 'text', this.TEXT_OUTPUT);
+    }
+
+    async getTranscriptTextStream(id: string): Promise<Readable> {
+        return await this.getOutputsHelper(id, 'stream', this.TEXT_OUTPUT);
+    }
+
+    async getCaptions(id: string): Promise<Readable> {
+        return await this.getOutputsHelper(id, 'stream', this.SRT_OUTPUT);
+    }
+
+    private async getOutputsHelper(id: string, type: any, format: string): Promise<any> {
```

Comment:
`type` can also be declared as a string enum

Reply:
Looking at this some more: If you use this all purpose output helper, you should specify the return type as `Promise<Readable|string|RevAiApiTranscript>`. The `Promise<Readable|string|RevAiApiTranscript>` return type is ugly, but more well defined than `Promise<any>`

I believe, if you make the above suggestions you will be forced to cast the output of `getOutputsHelper` to one of the three possibilities in each of the convenience wrappers: getCaptions, getTranscriptText, etc.

I do worry about future sdk developers misusing the `getOutputsHelper` function and passing in incompatible types and formats.

Reply:
What do you think?

Reply:
I didn't realize this. I think that makes sense

Reply:
The method is marked as private so they shouldn't be doing that. I don't think we should plan for customers using the SDK incorrectly and I also don't think we will run into any problems if they pass incorerct combinations. They'll just get errors from the api

Reply:
re:
> I do worry about future sdk developers misusing the getOutputsHelper function and passing in incompatible types and formats.

Here I am referring to future Rev developers, not to customer developers. That the function is private does not prevent rev developers from making mistakes.

Reply:
You don’t have to address anything in this PR. Just keep this in mind for when we extract an api wrapper.